### PR TITLE
Add ability to burn papers +Flammable tweaks.

### PR DESCRIPTION
**Pull request by mhamsterr**
Let it burn! Revival of #10373 
Most of the code was done by [JustinTrotter](https://github.com/JustinTrotter) in his pr and then modified by me (JustinTrotter, please leave a comment here / find me on discord if you want me to close this pr / change something). 

## About the PR

+ Now you can specify if entity should ignite because of contact with isHot entities.
+ Now you can specify the amount of firestacks that will be applied to entity when ignited.
+ Now you can specify if entity will lose it's firestacks because of interaction with other entities.

+ Papers can now burn to ash.
  + But fire cannot spread between them or any other items next to them

**Media**

https://github.com/space-wizards/space-station-14/assets/81412348/b5aae678-9aa3-4a62-8345-044aa61a022f

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: JustinTrotter & mhamster
- tweak: Papers now burn and turn to ash.

### CODE REVIEW

**Review by metalgearsloth — DISMISSED** [review]

File: Content.Server/Atmos/Components/FlammableComponent.cs [hunk 1]
```diff
@@ -35,5 +35,26 @@ public sealed class FlammableComponent : Component
...
+        [DataField("extinguishable")]
+        public bool CanExtinguish { get; private set; } = true;
+
+        /// <summary>
+        ///     How many firestacks should be applied to component when being set on fire?
+        /// </summary>
+        [ViewVariables(VVAccess.ReadWrite)]
+        [DataField("fireOnIgnite")]
+        public float FirestacksOnIgnite { get; private set; } = 2.0f;
     }
```
[the remark this answers is not in the document]
Comment (mhamsterr) [reply Content.Server/Atmos/Components/FlammableComponent.cs]: Like that in last commit? (i am at very low level at c# actually, sorry)

**Review by AJCM-git — CHANGES_REQUESTED** [review]

**Review by Emisse — APPROVED** [review]

**Comment by UbaserB** [comment]
walking over it should set you on fire

**Comment by mhamsterr** [comment]
> walking over it should set you on fire

sounds fair. will try to implement it before merge (if it is planned soon)

**Comment by mhamsterr** [comment]
> > walking over it should set you on fire
> 
> sounds fair. will try to implement it before merge (if it is planned soon) (and will do another pr if i don't make it on time)

actually nevermind, will do it in a separate pr

**Review by AJCM-git — APPROVED** [review]

**Pull request merged**